app/src/main/java/com/globant/harrypotterapp/fragment/CharactersFragment.kt

```diff
@@ -58,7 +61,9 @@ class CharactersFragment : Fragment() {
         Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
     }
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? = binding.root
+    override fun showCharacterDetail() {
+        this.context?.let { startActivity(CharacterDetailActivity.getIntent(it)) }
```

Comment:
this should call the viewModel to set the new state of this liveData.. then you will call the startActivity in the fragment

Reply:
Done!

Reply:
this still here

Reply:
Sorry Mau, refactored!